### ISSUE TITLE
502 Bad Gateway

### ISSUE THREAD

**Issue by secret-weapon** [facebook/fbctf#23]
run url= https://localhost

server response :
502 Bad Gateway nginx.
what is the problem? :(

**Comment by melona380** [comment]
Same Issue... ANY SOLUTION?

**Comment by nicholasykl** [comment]
I would like to know the solution too :(

**Comment by mpomarole** [comment]
Could you post more details of what you ran to provision ? And please provide a past of the logs so we can help you better.

Instructions to provision for development mode are at: https://github.com/facebook/fbctf#development

**Comment by rverton** [comment]
I had this problem when trying to install it on a 14.04 32-bit ubuntu. hhvm fails to install (64bit required) and so the gateway nginx tries to connect to (127.0.0.1:9000) is not running (thus the 502 Bad Gateway error). Can you check if hhvm is installed and if there is a port 9000 opened?

**Comment by nicholasykl** [comment]
I believe I have this solved by using the 64bit version of Ubuntu 14.04. 
I clean installed (easy install) the OS on the 64bit VMware Player, and didn't update any pre-installed packages. I then follow the readme instructions to setup the DEV environment.

**Comment by NeoVilsonWong** [comment]
@mpomarole new error 

[502error.txt](https://github.com/facebook/fbctf/files/262098/502error.txt)

**Comment by mpomarole** [comment]
@NeoVilsonWong you are still having the problem of locale not being set correctly. Did you try http://askubuntu.com/questions/162391/how-do-i-fix-my-locale-issue ?

Specially adding `export LC_ALL="en_US.UTF-8"` to your .bashrc file?

**Comment by NeoVilsonWong** [comment]
@mpomarole  done check this 
<img width="254" alt="screen shot 2016-05-13 at 3 28 50 am" src="https://cloud.githubusercontent.com/assets/7342012/15227654/de8d0df8-18ba-11e6-929f-deed88b03b63.png">

**Comment by mpomarole** [comment]
@NeoVilsonWong what you have below:

```
WARNING! Your environment specifies an invalid locale.
 This can affect your user experience significantly, including the
 ability to manage packages. You may install the locales by running:

   sudo apt-get install language-pack-UTF-8
     or
   sudo locale-gen UTF-8

To see all available language packs, run:
   apt-cache search "^language-pack-[a-z][a-z]$"
To disable this message for all users, run:
   sudo touch /var/lib/cloud/instance/locale-check.skip
```

Is clearly what is failing to install all the proper packages on the provision script.

**Comment by melona380** [comment]
Hi, I found a temporary solution and it's not a permanent solution but it will get you up and running for now.
I was trying to run the development version on Ubuntu 14.04 and I was getting the 502 Gateway timeout. Using the argumnet 'dev' Most people are using the development version just because it auto generates the SSL certificates. What I did to fix it was
1. Install fresh Ubuntu 14.04
2. Generate self-signed certs by:
        A) openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout  /dir/to/ssl/fbctf.enckey -out /dir/to/ssl/fbctf.crt
        B) openssl rsa -in /dir/to/ssl/fbctf.enckey -out /dir/to/ssl/fbctf.key
3.run the install with the 'prod' flag
4. Enter path to sslcerts during install
5. Enjoy

Note if you want free ssl that is from a real ca go to startssl.com and you will have ssl in a day or so. For FREE, if your cheap like me:)
